tests/system/Debugger/E2ETest.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Copyright 2018 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Tests\System\Debugger;
+
+use Google\Cloud\Debugger\V2\Gapic\Debugger2GapicClient as GapicClient;
+use Google\Cloud\Debugger\V2\Breakpoint;
+use Google\Cloud\Debugger\V2\SourceLocation;
+use Google\Cloud\TestUtils\EventuallyConsistentTestTrait;
+use Google\Cloud\TestUtils\AppEngineDeploymentTrait;
+use GuzzleHttp\Client;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * The test deploys the sample app contained in the app folder to Google App
+ * Engine Flexible Environment. Before deployment, we generate a composer.json
+ * that uses the current branch of google/cloud to test against.
+ *
+ * Each test sets a breakpoint, ensures that the app has seen the breakpoint,
+ * then makes a request to the app that should trigger the breakpoint. We then
+ * ensure that the breakpoint has been fulfilled.
+ *
+ * @group debugger
+ */
+class E2ETest extends TestCase
+{
+    protected static $debuggeeId;
+    protected static $httpClient;
+
+    use AppEngineDeploymentTrait;
+    use EventuallyConsistentTestTrait;
+
+    public static function beforeDeploy()
+    {
+        self::createComposerJson();
+        self::$gcloudWrapper->setDir(implode(DIRECTORY_SEPARATOR, [__DIR__, 'app']));
+    }
+
+    public static function afterDeploy()
+    {
+        $url = self::$gcloudWrapper->getBaseUrl();
+        self::$httpClient = new Client(['base_uri' => $url]);
+
+        $resp = self::$httpClient->get('/debuggee');
+        $data = json_decode($resp->getBody()->getContents(), true);
+        self::$debuggeeId = $data['debuggeeId'];
+    }
+
+    public static function tearDownAfterClass()
+    {
+        self::deleteApp();
+    }
+
+    public static function createComposerJson()
+    {
+        $branch = exec('git rev-parse --abbrev-ref HEAD');
```

Comment:
Note that this is parsing the repo/branch from the local git repo. Is there another preferred method for grabbing this?

I assume that we only run these system tests from a development machine rather than as part of CI.

Reply:
Soon we will run the system tests in a cron job on travis. This looks to have some relevant information as to solutions: https://graysonkoonce.com/getting-the-current-branch-name-during-a-pull-request-in-travis-ci/

Reply:
Ok, I can future proof it. I was unsure if we planned to use travis or circle and didn't want to add both.